packages/sanity/src/ui/dialog/dialog.tsx

```diff
@@ -0,0 +1,53 @@
+import {Button, Flex, Dialog as UIDialog, DialogProps as UIDialogProps, Box} from '@sanity/ui'
+import React, {ComponentProps, forwardRef} from 'react'
+
+/** @internal */
+export interface DialogProps extends Omit<UIDialogProps, 'footer'> {
+  children?: React.ReactNode
+  footer?: {
+    // TODO: When `@sanity/ui` has `Button` component, use those props instead.
+    cancelButton?: ComponentProps<typeof Button>
+    confirmButton?: ComponentProps<typeof Button>
+  }
+}
+
+/**
+
+ * Studio UI <Dialog>.
+ *
+ * Studio UI components are opinionated `@sanity/ui` components meant for internal use only.
+ * Props and options are intentionally limited to ensure consistency and ease of use.
+ *
+ * @internal
+ */
+export const Dialog = forwardRef(function Dialog(
+  {footer, children, ...props}: DialogProps,
+  ref: React.Ref<HTMLDivElement>,
+) {
+  return (
+    <UIDialog
+      {...props}
+      ref={ref}
+      footer={
+        (footer?.confirmButton || footer?.cancelButton) && (
+          <Flex width="full" gap={3} justify="flex-end" padding={3}>
+            {props.onClose && (
+              <Button
+                mode="bleed"
+                text="Cancel"
+                tone="default"
+                onClick={props.onClose}
+                {...footer.cancelButton}
+              />
+            )}
+            {footer.confirmButton && (
+              <Button mode="default" text="Confirm" tone="critical" {...footer.confirmButton} />
+            )}
+          </Flex>
+        )
+      }
+    >
+      <Box paddingX={3}>{children}</Box>
```

Comment:
Similarly, the default padding for children should accommodate all use cases within the studio. Ideally, `children` of the Studio UI dialog shouldn't need to specify any custom padding at all. This custom padding value should ensure that child text are horizontally aligned with header text.

Reply:
Upon further reflection, there are cases where it makes sense for dialog `children` to be completely flush within the body of the dialog (e.g. document inspect, default asset source – cases where there's sticky elements and/or a need for x-axis overflow).

Perhaps Studio UI tooltip `children` could always default to specific padding to cover the vast majority of cases (confirming / warning dialogs, editing document forms) whilst we provide an escape hatch for users to bypass this? 

Ideally, studio developers shouldn't even need to think about the proper padding required for dialog `children`, barring unique cases

Reply:
Hey @robinpyon thanks for this review.
The `paddingX={3}` in the body is coming from the design file.
If you check that, all the dialogs displayed have 2 different paddings in the body.
The first one encapsulates the body in a padding 3, to make it match with the header and footer padding. 

<img width="726" alt="Screenshot 2023-10-30 at 16 14 31" src="https://github.com/sanity-io/sanity/assets/46196328/03ca1be6-20f4-4182-84a4-4ce6120f7bfa">

Then, depending on the content it has more or less padding, but always keeping the initially defined padddingX=3
E.g 
- The ConfirmDeleteDialog has a `paddingX=3` here.
<img width="758" alt="Screenshot 2023-10-30 at 16 14 45" src="https://github.com/sanity-io/sanity/assets/46196328/80f26631-0e36-4cc1-9e16-40f50b4a5708">

-  Form dialogs use a `paddingX=5` 
<img width="720" alt="Screenshot 2023-10-30 at 16 21 26" src="https://github.com/sanity-io/sanity/assets/46196328/abdb0646-fee4-4578-81b7-2ba5bb21ea89">